api/v1/runtime/common.proto

```diff
@@ -44,9 +44,20 @@ message Tuple {
   uint32 dst_port = 4;
 }
 
+enum FlowDirection {
+  FLOW_DIRECTION_UNKNOWN = 0;
+  FLOW_DIRECTION_INCOMING = 1;
+  FLOW_DIRECTION_OUTGOING = 2;
+}
+
+message NetworkMetadata {
+  FlowDirection flow_direction = 1;
+}
+
 message DNS {
   string DNSQuestionDomain = 1;
   repeated DNSAnswers answers = 2;
+  NetworkMetadata networkMetadata = 3;
```

Comment:
Doesn't look like we can much more fields here. I would put this as direct flow_direction on DNS message without extra nesting because now it will allocate additional pointer for each dns event.